hedgehog/src/Hedgehog/Internal/Runner.hs

```diff
@@ -126,11 +127,20 @@ checkReport ::
   -> Test m ()
   -> (Report -> m ())
   -> m Report
-checkReport cfg size0 seed0 test0 updateUI =
-  let
-    test =
-      catchAll test0 (fail . show)
+checkReport cfg size0 seed0 test updateUI =
+  checkReportNoCatch cfg size0 seed0 (catchAll test (fail . show)) updateUI
 
+checkReportNoCatch ::
+     forall m.
+     MonadIO m
+  => PropertyConfig
+  -> Size
+  -> Seed
+  -> Test m ()
+  -> (Report -> m ())
+  -> m Report
+checkReportNoCatch cfg size0 seed0 test updateUI =
```

Comment:
I don't think this will do what you want, you'll end up checking that any of the 100 or so tests throws an exception, but it sounds like you want to check that all of the tests throw an exception